cumulusci/tasks/bulkdata/snowfakery.py

```diff
@@ -0,0 +1,657 @@
+# MAX_PORTION_SIZE. This is to prevent the org wasting time waiting
+# for the first portions.
+MIN_PORTION_SIZE = 2_000
+MAX_PORTION_SIZE = 250_000
+ERROR_THRESHOLD = (
+    0  # TODO v2.1: Allow this to be a percentage of recent records instead
+)
+
+# time between "ticks" where the org re-evaluates its progress
+# relatively arbitrary trade-off between busy-waiting and adding latency.
+WAIT_TIME = 3
+
+
+class Snowfakery(BaseSalesforceApiTask):
+
+    task_docs = """
+    Do a data load with Snowfakery.
+
+    All options are optional.
+
+    The most commonly supplied options are `recipe` and one of the three
+    `run_until_...` options.
+    """
+
+    task_options = {
+        "recipe": {
+            "required": True,
+            "description": "A Snowfakery recipe file determining what data to generate and load.",
+        },
+        "run_until_records_in_org": {
+            "description": """<sobject>:<count>
+
+      Run the recipe repeatedly until the count of <sobject>
+      in the org matches the given <count>.
+
+      For example, `--run_until_records_in_org Account:50_000` means:
+
+      Count the Account records in the org. Let’s say the number
+      is 20,000. Thus, we must run the recipe over and
+      over again until we generate 30,000 new Account records.
+      If the recipe also generates e.g.Contacts, Opportunities or whatever
+      else, it generates the appropriate number of them to match.
+
+      Underscores are allowed but optional in big numbers: 2000000
+      is the same as 2_000_000.
+        """
+        },
+        "run_until_records_loaded": {
+            "description": """<sobject>:<count>
+
+      Run the recipe repeatedly until the number of records of
+      <sobject> uploaded in this task execution matches <count>.
+
+      For example, `--run_until_records_loaded Account:50_000` means:
+
+      Run the recipe over and over again
+      until we generate 50_000 new Account records. If the recipe
+      also generates e.g. Contacts, Opportunities or whatever else, it
+      generates the appropriate number of them to match.
+        """
+        },
+        "run_until_recipe_repeated": {
+            "description": """Run the recipe <count> times,
+            no matter what data is already in the org.
+
+            For example, `--run_until_recipe_repeated 50_000` means
+            run the recipe 50_000 times."""
+        },
+        "working_directory": {
+            "description": "Default path for temporary / working files"
+        },
+        "loading_rules": {
+            "description": "Path to .load.yml file containing rules to use to "
+            "load the file. Defaults to `<recipename>.load.yml`. "
+            "Multiple files can be comma separated."
+        },
+        # "recipe_options": {},  # TODO: Snowfakery 2.1
+        "bulk_mode": {
+            "description": "Set to Serial to force serial mode on all jobs. Parallel is the default."
```

Comment:
Should this say something about the Bulk API in particular since the task will still be loading multiple portions in parallel, even when bulk_mode is set to serial

Reply:
@davisagli : What do yo think of this? I will make a new PR where bulk_mode = Serial implies only one uploader worker.

Reply:
@prescod Is that what we want? The current option is mainly there to serve the purpose of avoiding row lock errors (at the expense of speed). I think that's orthogonal to whether the user wants to load multiple portions in parallel or not. (I guess maybe not entirely, since multiple portions can have lookups to singletons which could be locked.)

If it is what we want, then `bulk_mode` would no longer be a good name for the option.